config/module.go

```diff
@@ -95,3 +92,20 @@ func (m Module) Equals(other Module) bool {
 	//nolint:govet
 	return reflect.DeepEqual(m, other)
 }
+
+// CopyOnlyPublicFields copies only exported fields on the module config.
+func (m *Module) CopyOnlyPublicFields() (*Module, error) {
+	// We're using JSON as an intermediary to ensure only the json exported fields are
+	// copied.
+	tmpJSON, err := json.Marshal(m)
+	if err != nil {
+		return nil, errors.Wrap(err, "error marshaling config")
+	}
+	var cfg Module
+	err = json.Unmarshal(tmpJSON, &cfg)
+	if err != nil {
+		return nil, errors.Wrap(err, "error unmarshaling config")
+	}
+
+	return &cfg, nil
+}
```

Comment:
I find this to be terribly ugly (so much so) that it might remove the entire point of this PR and be worse than things were before this.

I am open to suggestions.

Reply:
It's not great, but it is certainly a pattern I've seen in the RDK and elsewhere. I am wondering why you couldn't just use the `config.Module` passed into `Add`, though. It's not a pointer parameter, so I assumed it was copied already. Did you find that retaining the values of the `alreadyValidated` and `cachedErr` fields was causing weird behavior?

Reply:
retaining `alreadyValidated` and `cachedErr` was causing the deep equality checks to fail. :/

Reply:
Talked offline. Swapped to fixing the local_robot TestConfig test rather than working around it.

Resolving.